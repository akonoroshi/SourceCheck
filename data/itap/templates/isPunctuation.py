import string

def isPunctuation(s):
    return s in string.punctuation
