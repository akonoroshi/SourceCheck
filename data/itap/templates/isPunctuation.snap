Module(list {
    Import(list {
        alias:string(null)
    })
    FunctionDef:isPunctuation(arguments(list {
        arg:s(null)
    }, null, list, list, null, list), list {
        @defBlock CONDITION {
            Compare(Name:s(Load), list {
                In
            }, list {
                Attribute:punctuation(Name:string(Load), Load)
            })
        }
        @or {
            Return(@block CONDITION)
            If(@block CONDITION, list {
                Return(NameConstant:True)
            }, list {
                Return(NameConstant:False)
            })
        }
    }, list, null)
})
